draft-ietf-ace-oscore-profile.xml

```diff
@@ -546,7 +612,7 @@ OSCORE_Security_Context = {
       <section anchor="c-rs" title="C-to-RS: POST to authz-info endpoint">
 
         <t>
-          The client MUST use CoAP and the Authorization Information resource as described in section 5.8.1 of <xref target="I-D.ietf-ace-oauth-authz"/> to transport the token to the RS.
+          The client MUST generate a nonce N1 with a good amount of randomness. This profile RECOMMENDS to use a nonce of 64 bits. The client MUST store this nonce as long as the response from the RS is not received and the access token related to it is still valid. The client MUST use CoAP and the Authorization Information resource as described in section 5.8.1 of <xref target="I-D.ietf-ace-oauth-authz"/> to transport the token and N1 to the RS. The client MUST use the Content-Format "application/ace+cbor" defined in section 8.14 of <xref target="I-D.ietf-ace-oauth-authz"/>. The client MUST include the access token using the correct CBOR label (e.g., "cwt" for CWT, "jwt" for JWT) and N1 using the 'nonce' parameter defined in section 5.1.2 of <xref target="I-D.ietf-ace-oauth-authz"/>.
         </t>
```

Comment:
What is the reason for saying a 64-bit value?  A 16-bit values is much smaller and should be able to have the same type of ability to generate new session keys.  If one uses an "unpredictable counter" that is 16-bit long then I think it should be fine.

Reply:
ok, will change to that.

Reply:
Done in 1e11f5d

Reply:
Could you expand on what you mean by "unpredictable counter"? Is that just a random number? Why using the term "counter"? That brings us to think that it is not a random number, which may be a problem, for example if both client and server restart and use the same access token.

Reply:
After discussing with others, reverted back to 64bit random. Yes, the requirement is non re-use, a easy way to have that is by having a random nonce (and a big enough range to not have collisions). I also added some text in the sec cons about that.

Reply:
So if this is not a 64bit value is it rejected as a bad message?